shop/src/test/java/ro/msg/learning/shop/SingleStrategyIntegratedTest.java

```diff
@@ -0,0 +1,127 @@
+package ro.msg.learning.shop;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.util.NestedServletException;
+import ro.msg.learning.shop.dto.CustomerDto;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.dto.StockDto;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK,
+        classes = ShopApplication.class)
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:application-test-single.properties")
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+public class SingleStrategyIntegratedTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private LocationRepository locationRepository;
+    @Autowired
+    private StockRepository stockRepository;
+    private OrderDto orderDto;
+    private Customer customer;
+    private OrderDto orderDtoFailure;
+
+    @Before
+    public void setUpTest() throws Exception {
+        mockMvc.perform(post("/data/populate")).andExpect(status().isOk());
```

Comment:
It's a little hard to read the code here. Try to use empty lines to segmentate the logic into let's call them pieces